app/Actions/Update/Apply.php

```diff
@@ -59,94 +60,105 @@ private function check_prod_env_allow_migration(array &$output)
 	/**
 	 * call composer over exec.
 	 *
-	 * @param array $output
-	 */
-	private function call_composer(array &$output)
-	{
-		if (Configs::get_value('apply_composer_update', '0') == '1') {
-			// @codeCoverageIgnoreStart
-			Logs::warning(__METHOD__, __LINE__, 'Composer is called on update.');
-
-			// Composer\Factory::getHomeDir() method
-			// needs COMPOSER_HOME environment variable set
-			putenv('COMPOSER_HOME=' . base_path('/composer-cache'));
-			chdir(base_path());
-			exec('composer install --no-dev --no-progress --no-suggest 2>&1', $output);
-			chdir(base_path('public'));
-		// @codeCoverageIgnoreEnd
-		} else {
-			$output[] = 'Composer update are always dangerous when automated.';
-			$output[] = 'So we did not execute it.';
-			$output[] = 'If you want to have composer update applied, please set the setting to 1 at your own risk.';
-		}
-	}
-
-	/**
-	 * Arrayify a string and append it to $output.
+	 * @param string[] $output the per-line console output
 	 *
-	 * @param $string
-	 * @param array $output
+	 * @return void
 	 *
-	 * @return array
+	 * @throws FrameworkException
 	 */
-	private function str_to_array($string, array &$output)
+	private function call_composer(array &$output): void
 	{
-		$a = explode("\n", $string);
-		foreach ($a as $aa) {
-			if ($aa != '') {
-				$output[] = $aa;
+		try {
+			if (Configs::get_value('apply_composer_update', '0') == '1') {
+				// @codeCoverageIgnoreStart
+				Logs::warning(__METHOD__, __LINE__, 'Composer is called on update.');
+
+				// Composer\Factory::getHomeDir() method
+				// needs COMPOSER_HOME environment variable set
+				putenv('COMPOSER_HOME=' . base_path('/composer-cache'));
+				chdir(base_path());
+				exec('composer install --no-dev --no-progress --no-suggest 2>&1', $output);
```

Comment:
https://getcomposer.org/upgrade/UPGRADE-2.0.md
> Deprecated --no-suggest flag as it is not needed anymore
```suggestion
				exec('composer install --no-dev --no-progress 2>&1', $output);
```

Reply:
Again, not my code. I only copied it or more precisely, wrapped it into an exception handler. That's why it appears as new code.